### PR TITLE
Paul imagemin 1

### PR DESCRIPTION
**Pull request by starsinmypockets**
Refactor imagemagick lib to do minifcation using imagemin if flag is present
Add tests

### CODE REVIEW

File: test-imagecrunch.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ var _ = require('lodash');
 
 // Test the imagecrunch image backend, written specifically for Macs
 
-var localOptions = { storage: 'local', image: 'imagecrunch', uploadsPath: __dirname + '/test', uploadsUrl: 'http://localhost:3000/test' };
+var localOptions = { storage: 'local', backend: 'imagecrunch', uploadsPath: __dirname + '/test', uploadsUrl: 'http://localhost:3000/test' };
```
Comment (boutell) [inline test-imagecrunch.js]: This looks like it reverted to something older?

File: test-imagemagick.js [hunk 2]
```diff
@@ -0,0 +1,193 @@
+var uploadfs = require('./uploadfs.js')();
```
Comment (boutell) [inline test-imagemagick.js]: Not sure why this is new here... was this inadvertently removed previously? I agree it is not a priority to mocha-ize it etc., please don't do that right now.

File: lib/image/imagemagick.js [hunk 3]
```diff
@@ -367,7 +382,63 @@ module.exports = function() {
...
+            })
+            .catch(err => {
+              console.log('ERROR', err);
+              // returning ok callback as this additionnal compression doesn't prevent images usage
+              return callback(err);
+            });
+        });
+      }
+
+      function convertSvg (context, callback) {
```
Comment (boutell) [inline lib/image/imagemagick.js]: This doesn't crop, so it's not really a fully valid uploadfs implementation for svg files. I'll talk to Ludo about whether he is using this and considers it important to release before we add time for that.

File: test-imagecrunch.js (hunk 1 above)
Comment (starsinmypockets) [reply test-imagecrunch.js]: fixed

File: lib/image/imagemagick.js (hunk 3 above)
Comment (starsinmypockets) [reply lib/image/imagemagick.js]: Agreed. Let me know how to proceed. I guess the options are to remove this, or to implement, or leave the hack so it doesn't break existing implementations...

Comment (boutell) [reply lib/image/imagemagick.js]: Please remove svg for now so we can get this out the door to npm, however I have opened a new ticket to bring back the svg support plus cropping to match other formats.

**Pull request merged**